charms/layers/java/lib/openjdk.py

```diff
@@ -11,6 +12,10 @@ def installopenjdk():
     conf = hookenv.config()
     install_type = conf['install-type']
     java_major = conf['java-major']
+    #openjdk 8 is not included in ubuntu repos    
+    if java_major == '8':
+        charms.apt.add_source('ppa:openjdk-r/ppa')
+        subprocess.check_call(['apt-get','update'])
```

Comment:
Don't run `apt-get` manually, use the apt layer. More info: https://github.com/stub42/layer-apt

*Note: I don't think it's nessesary to run `apt-get` at all, see from the apt layer docs:*

> add_source(source, key=None) ... **Sets the apt.needs_update reactive state.**